Services/Administration/OJS.Services.Administration.Business/Problems/ProblemsBusinessService.cs

```diff
@@ -360,21 +366,37 @@ await this.transactionsProvider.ExecuteInTransaction(async () =>
         await this.submissionsCommonBusinessService.PublishSubmissionsForProcessing(serviceModels);
     }
 
-    private static void AddSubmissionTypes(Problem problem, ProblemAdministrationModel model)
+    private async Task AddSubmissionTypes(Problem problem, ProblemAdministrationModel model)
     {
-        var newSubmissionTypes = model.SubmissionTypes
-            .Select(x => new SubmissionTypeInProblem
+        var submissionTypeIds = model.SubmissionTypes.Select(st => st.Id).ToList();
+
+        var submissionTypes = await this.submissionTypesData
+            .All(st => submissionTypeIds.Contains(st.Id))
+            .ToDictionaryAsync(st => st.Id, st => st);
```

Comment:
You can make a `HashSet` from the ids instead of a dictionary, as you don't use the `value`, just the `key`.

Reply:
There is no `.ToHashSetAsync()` method, this is why I used `.ToListAsync()`, the number of `SubmissionTypes` is not large, so the filtration with a `list` should not be slow.

Reply:
I know, but you can create it from list. However, as you said, the collection is so small, there will probably not be any benefit of the HashSet, maybe even it would be worse, due to creation of the HashSet.